static/src/javascripts/projects/common/modules/navigation/newSticky.js

```diff
@@ -0,0 +1,80 @@
+define([
+    'fastdom',
+    'common/utils/$',
+    'common/utils/mediator',
+    'lodash/functions/bindAll',
+    'common/utils/detect'
+], function (
+    fastdom,
+    $,
+    mediator,
+    bindAll,
+    detect) {
+
+    function StickyHeader() {
```

Comment:
Why not just have the init method's content in the constructor?

Then you could just define your module as

    function StickyHeader() {
         setup();
    }

    return new StickyHeader();

This way you wouldn't have to know if newSticky had already been initialized when you grab the service - you wouldn't be splitting definition and generation. In your test code you could just do

     let stickyHeader;
     beforeEach(()=> {
         stickyHeader = new stickyHeaderService.constructor();
     });